vispy/visuals/graphs/util.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015, Vispy Development Team.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+"""
+Graph utilities
+===============
+
+A module containing several graph utility functions.
+"""
+
+import itertools
+
+import numpy as np
+
+try:
+    from scipy.sparse import issparse
+except ImportError:
+    def issparse(*args, **kwargs):
+        return False
+
+
+def get_edges(adjacency_mat):
+    if issparse(adjacency_mat):
+        func = _sparse_get_edges
+    else:
+        func = _get_edges
+
+    for result in func(adjacency_mat):
+        yield result
+
+
+def _sparse_get_edges(adjacency_mat):
+    # Each non zero value in a COO matrix is stored in (row, col, value)
+    # format
+    coo_mat = adjacency_mat.tocoo()
+    return zip(coo_mat.row, coo_mat.col)
+
+
+def _get_edges(adjacency_mat):
+    # Get indices of all non zero values
+    i, j = np.where(adjacency_mat)
+
+    return zip(i, j)
+
+
+def get_directed_edges(adjacency_mat):
+    if issparse(adjacency_mat):
+        func = _sparse_get_directed_edges
+    else:
+        func = _get_directed_edges
+
+    for result in func(adjacency_mat):
+        print("Directed", result)
+        yield result
+
+
+def _sparse_get_directed_edges(adjacency_mat):
+    iu = np.triu_indices_from(adjacency_mat, 1)
+    il = np.tril_indices_from(adjacency_mat)
+
+    lower = adjacency_mat.copy()
+    lower[il] = 0
+
+    adjacency_mat[iu] = 0
+
+    return itertools.chain(
+        _sparse_get_edges(adjacency_mat),
+        _sparse_get_edges(lower)
+    )
+
+
+def _get_directed_edges(adjacency_mat):
+    upper = np.triu(adjacency_mat, k=1)
+    lower = np.tril(adjacency_mat)
+
+    return itertools.chain(get_edges(upper), get_edges(lower))
```

Comment:
in the `_sparse` version you call the `_sparse_get_edges`, but here you call `get_edges`. Either both should call the specific version, or both should call the generic version.